MsWheaPkg/Test/UnitTests/MsWheaEarlyStorageUnitTestApp/MsWheaEarlyUnitTestApp.c

```diff
@@ -601,15 +601,41 @@ MsWheaEarlyUnitTestAppEntryPoint (
     NULL
     );
 
-  AddTestCase (
-    Misc,
-    "Free ES slot find",
-    "MsWhea.Miscellaneous.MsWheaESFindSlotTest",
-    MsWheaESFindSlotTest,
-    MsWheaESVerify,
-    MsWheaESCleanUp,
-    NULL
-    );
+  // Only run the following tests if the early storage capacity is large enough
+  if (MsWheaEarlyStorageGetMaxSize () >= (sizeof (MS_WHEA_EARLY_STORAGE_HEADER) + (sizeof (MS_WHEA_EARLY_STORAGE_ENTRY_V0)))) {

```

Comment:
Are you going by the actual value of `PcdMsWheaReportEarlyStorageCapacity` as assigned in the package DSC building the host-based unit test?

If so, I think that should be mocked and positive and negative cases tested.

Reply:
What do you mean by "mocked and positive and negative cases tested?"

Reply:
Write mock instances of `MsWheaEarlyStorageGetMaxSize()`.

- **Mock 1**: Return a size less than: `(sizeof (MS_WHEA_EARLY_STORAGE_HEADER) + (sizeof (MS_WHEA_EARLY_STORAGE_ENTRY_V0))`
- **Mock 2**: Return a size equal to: `(sizeof (MS_WHEA_EARLY_STORAGE_HEADER) + (sizeof (MS_WHEA_EARLY_STORAGE_ENTRY_V0))`
- **Mock 3**: Return a size greater than: `(sizeof (MS_WHEA_EARLY_STORAGE_HEADER) + (sizeof (MS_WHEA_EARLY_STORAGE_ENTRY_V0))`

Verify functions dependent on `MsWheaEarlyStorageGetMaxSize()` behave as expected given the value returned from `MsWheaEarlyStorageGetMaxSize()`.

Reply:
I think it's best to just log an error if the early storage region is not large enough to store at least one ES entry. The test which requires two entry slots is still blocked behind a check against the ES capacity but will not log an error if it is too small.

Reply:
Sounds reasonable